src/views/Detail.vue

```diff
@@ -76,31 +76,15 @@
           v-for="(item, index) in items"
           :key="index"
         >
-          <ImageDetail
-            v-if="fileType === Types.images"
-            :file="item"
-          />
-          <DirectoryDetail
-            v-if="fileType === Types.directories"
-            :file="item"
-          />
-          <DocumentDetail
-            v-if="fileType === Types.text"
-            :file="item"
-          />
-          <AudioDetail
-            v-if="fileType === Types.audio"
-            :file="item"
-          />
-          <VideoDetail
-            v-if="fileType === Types.video"
+          <component
+            :is="DetailComponent[this.fileType]"
             :file="item"
           />
         </v-carousel-item>
       </v-carousel>
       <component
         v-else
-        :is="componentType"
+        :is="DetailComponent[this.fileType]"
```

Comment:
I previously asked you to remove this because the original is higher in verbosity and because it is virtually impossible to find documentation on the 'component'-component (for name reasons). At the very least, please at a comment to the reference of `<component>`.

Reply:
Well to me, the current version seems very clear and concise, so it didn't seem necessary to me to add syntax explanation to the comments. But since you keep insisting I will add a line with reference to the documentation.

Reply:
btw you should search on the word 'is' instead https://vuejs.org/v2/guide/components.html#Dynamic-Components

Reply:
It is cluttery to put references to documentation about common code patterns in the code, typically, unless it is really arcane or some hacky solution from stackoverflow. Code should speak for itself as much as possible.

Reply:
@dokterbob I don't see the change, but I did push it!